shared/middleware/helpers/dynamo_table.rb

```diff
@@ -0,0 +1,359 @@
+require 'aws-sdk'
+
+#
+# DynamoTable
+#
+class DynamoTable
+  MAX_BATCH_SIZE ||= 25
+  CHANNEL_TABLE_NAME_INDEX ||= "channel_id-table_name-index"
+
+  class NotFound < Sinatra::NotFound
+  end
+
+  def initialize(channel_id, storage_id, table_name)
+    _, @channel_id = storage_decrypt_channel_id(channel_id) # TODO(if/when needed): Ensure this is a registered channel?
+    @storage_id = storage_id
+    @table_name = table_name
+
+    @hash = "#{@channel_id}:#{@table_name}:#{@storage_id}"
+    @metadata_hash = "#{@channel_id}:#{@table_name}:#{@storage_id}:metadata"
+  end
+
+  def db
+    @@dynamo_db ||= Aws::DynamoDB::Client.new
+  end
+
+  def metadata
+    @metadata_item ||= db.get_item(
+        table_name: CDO.dynamo_table_metadata_table,
+        consistent_read: true,
+        key: {'hash'=>@metadata_hash}
+    ).item
+
+    # only return the parts we care about
+    @metadata_item.select{|k,_| k == "column_list"} if @metadata_item
+  end
+
+  def set_column_list_metadata(column_list)
+    db.put_item(
+      table_name: CDO.dynamo_table_metadata_table,
+      item: {
+        hash: @metadata_hash,
+        channel_id: @channel_id,
+        table_name: @table_name,
+        column_list: column_list.to_json,
+        updated_at: DateTime.now.to_s,
+      }
+    )
+
+    metadata
+  end
```

Comment:
When we insert a single row, we make some effort to not create a duplicate row id. When we create table metadata in dynamodb, I think we will just quietly overwrite whatever is there already. This seems ok. Does this sound right?